dashboard/app/models/level.rb

```diff
@@ -349,6 +351,27 @@ def strip_name
     self.name = name.to_s.strip unless name.nil?
   end
 
+  def log_changes
+    log = JSON.parse(audit_log || "[]")
+
+    # serialize 'properties' changes
+    changed = self.changed
+    if changed.include?('properties')
+      changed.delete('properties')
+      changed_attributes['properties'].each do |key, value|
+        changed.push(key) if properties[key] != value
+      end
+    end
```

Comment:
I don't totally understand this.  What does this accomplish?  And should we be cloning `changed` before we start messing with it, in case another hook depends on it too?